### ISSUE TITLE
Error: Cannot find module '../../../node_modules/use-sync-external-store/shim/index.js'

### ISSUE THREAD

**Issue by 1mihailsB** [nandorojo/solito#119]
Hello.

I was following this page https://github.com/nandorojo/solito/tree/master/example-monorepos/blank.
Downloaded example project. Ran it, it works both in web and native. Then tried to add react-query
```
cd packages/app
yarn add react-query
cd ../..
yarn
```
Wrapped everything in return() in packages/app/provider/index.tsx with 
<QueryClientProvider client={queryClient}></QueryClientProvider>
When starting the project I get the error `Error: Cannot find module '../../../node_modules/use-sync-external-store/shim/index.js'`

I am not a framework developer so I don't know how most things work under the hood. Maybe I'm trying to use a library (react-query) that isn't supposed to work in solito ? Where can I find which libraries should and which shouldn't work in such project ?

Thanks.
![image](https://user-images.githubusercontent.com/46363244/182267093-a73e5c32-be53-4cc4-8989-376ae92dfb69.png)

**Comment by nandorojo** [comment]
this is a react query issue, not a solito issue. you probably need to install `use-sync-external-store`. please reference react query’s docs and repo.

**Comment by 1mihailsB** [comment]
Thanks. You are right.

To avoid opening another issue I'll ask here:
sorry if I missed it in this repo docs or on solito website, but is there any description anywhere about what does a developer need to learn before starting to work with solito ? 
I know how to make a fullstack website with Next JS. Then I saw you on Theo's stream, watched your video about Beatgig and your stack and got an impression that I could make a website + app with your stack. First thing I tried with your solito example project is to make an Next JS in pages/api, return some data from it and fetch it both in web and native, from same packages/app file (detail-screen.tsx). Fetching this API works with react query when I start it as a web project, but not when I start it as native. Maybe `yarn native` doesn't start the next server same as `yarn web`, or w/e other reason it might be. Or maybe it just requires some special settings to be able to use localhost NextJS backend api from native app (› Scan the QR code above with Expo Go (Android) or the Camera app (iOS)).
And I don't really have an idea what to do about this, I couldn't find anything about data fetching on solito website or this repo.

This is just a very specific example but I imagine if I tried to do some other things that I'm used to doing in NextJS, I would end up in same situation as with data fetching.

So please, if I missed such information somewhere in your resource, please give me a hint where they are.
And if you didn't write such information for developers without experience with developing native apps or web+native(or react-native specifically) in single codebase, maybe it's worth to add somewhere on solito website or this repo what people like me (coming with only NextJS experience) should learn before trying to use your stack. And maybe some paragraph about what we should not expect to work from Next JS/react world in such stack for web+native development, and what would be the alternatives.

Thanks

**Comment by nandorojo** [comment]
api routes could be a page on the docs

https://github.com/axeldelafosse/expo-next-monorepo-example/issues/9#issuecomment-952062742
